TinyGPS.cpp

```diff
@@ -367,10 +386,19 @@ void TinyGPS::get_datetime(unsigned long *date, unsigned long *time, unsigned lo
 
 void TinyGPS::f_get_position(float *latitude, float *longitude, unsigned long *fix_age)
 {
-  long lat, lon;
-  get_position(&lat, &lon, fix_age);
-  *latitude = lat == GPS_INVALID_ANGLE ? GPS_INVALID_F_ANGLE : (lat / 1000000.0);
-  *longitude = lat == GPS_INVALID_ANGLE ? GPS_INVALID_F_ANGLE : (lon / 1000000.0);
+  get_lat_lon(latitude, longitude);
+
+  if (latitude) 
+  {
+    *latitude = (lat == GPS_INVALID_ANGLE) ? GPS_INVALID_F_ANGLE : (lat / 1000000.0);
+  }
+
+  if (longitude)
+  {
+    *longitude = (lat == GPS_INVALID_ANGLE) ? GPS_INVALID_F_ANGLE : (lon / 1000000.0);
```

Comment:
*longitude = (lon == GPS_INVALID_ANGLE) ? GPS_INVALID_F_ANGLE : (lon / 1000000.0);

"lon" instead of "lat", no?